section1/reflection.md

```diff
@@ -2,18 +2,43 @@
+
 2. What are the data types you learned about in this section? In your own words, define each.
+I learned about three primitive (fundamental) data types:
+  * String: letters (words), characters, and numbers wrapped in quotes
+  * Number: all positive and negative numbers including whole numbers and floating point numbers (decimals
+  * Boolean: true or false
 
 3. How would you log the string `"Hello World!"` to the console?
-
+```javaScript
+console.log(`Hello World!`);
+```
 4. What is/are the character(s) you would use to indicate comments in a JavaScript file? What is the purpose of a code comment?
+``// for a single-line comment``
+
+```javaScript
+/* for a
+multi-line comment
+like this one */
+```
 
 5. In your own words, what is a variable? How would you explain it to a 5 year old?
+A *variable* is a temporarily stored bit of data needed to do a job. It can change each time a script runs.
+A variable is like a plate. Your plate can be different colors and sizes just like the kind of food it holds can be different too. Some plates are better for eating some foods versus others, so you pick the best plate for the kind of food you want to eat!
 
 6. Think of a site or app you use frequently. What are three variables that are probably used? Which data type would each of those variables probably hold?
+  * username: script
+  * songLength: number
+  * clickPlayButton: boolean
 
 7. In your own words, explain what concatenation is.
+**Concatenation** is joining together separate variables with an operator, typically **+**.
 
 8. Think of a site or app you use frequently. Where do you think the developers used concatenation?
+  A welcome message telling me where I left off in my typing lesson.
 
 9. What questions do you still have about the work we've done so far? (not having a question is not an option)
+~~I have no questions.~~
+  * How do reassigned variables work? I did the exercises but do not fully understand how Javascript interprets that the variable should be reassigned if you are just retyping the variable name and assigning it a new name. I will research this independently as well.
+  * How does a boolean data type actually produce a viable piece of data. So far in the examples I've done, I haven't seen it produce a piece of data an end user could interpret in a meaningful way. Is that because it isn't meant to be seen by the end user and instead for a developer's code to run a certain way?
```

Comment:
Correct - end user doesn't probably ever see a "true" or "false"; rather something happens or doesn't happen based on their expectations and the situation.

Think of that red dot we see on so many apps that alerts us to a notification. I'll use Twitter. When I visit my Twitter app/website, some code runs and asks the question" does Amy have any notifications?" IF I do, the code that makes the red dot appear runs. Else - that code won't run, and I as a user don't see the red dot.